src/tail/mod.rs

```diff
@@ -19,12 +31,29 @@ impl Tail {
         let mut runtime = TokioRuntime::new()?;
 
         runtime.block_on(async {
+            // Create three [one-shot](https://docs.rs/tokio/0.2.16/tokio/sync#oneshot-channel)
+            // channels for handling ctrl-c. Each channel has two parts:
+            // tx: Transmitter
+            // rx: Receiver
```

Comment:
i may turn this inside out and create a "subscribe" method on the sigint handler but this works fine for now.